backend/src/main/java/com/now/naaga/gameresult/repository/GameResultRepository.java

```diff
@@ -3,8 +3,14 @@
 import com.now.naaga.gameresult.domain.GameResult;
 import java.util.List;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 
 public interface GameResultRepository extends JpaRepository<GameResult, Long> {
 
     List<GameResult> findByGameId(final Long gameId);
+
+    @Query("SELECT r FROM GameResult r JOIN FETCH r.game LEFT JOIN FETCH r.game.hints Where r.game.player.id = :playerId")
+    List<GameResult> findByPlayerId(@Param("playerId") Long playerId);
```

Comment:
P4:

좋은 것 같습니다.
`GameResult`<-> `Game`은 @OneToOne 연관관계이고
` Game`<-> `List<Hint>는 @OneToMany 연관계여서
다중 fetch join 문제도 딱히 없을 것 같습니다.

다만 jsql 가독성을 위해서 r.game 즉 Game에도 앨리어스를 주는 것은 어떨까요?
```suggestion
    @Query("SELECT r FROM GameResult r JOIN FETCH r.game g LEFT JOIN FETCH g.hints Where g.player.id = :playerId")
    List<GameResult> findByPlayerId(@Param("playerId") Long playerId);
```

Reply:
P3:

리뷰를 달 곳이 없어서 이곳에 달자면,
현재는 PR 변경사항에는 큰 영향은 끼치지 않지만,
전반적인 Game과 GameResult에 대해서 수정을 하면서 지연로딩을 걸어야할 것 같은데 어떻게 생각하시나요?

Reply:
가독성 수정 좋네요..! 반영하겠습니다~!
현재는 fetch join 으로 게임결과 조회시 게임과 힌트들을 모두 가져오고 있는데 그거와 별개로 게임결과를 조회할때 게임의 대한 정보가 필요 없는 상황이 있을 수 있으니 지연로딩을 거는것이 맞는것같네요! 의견 반영하겠습니다~!